Analysis/HistProducerSample.py

```diff
@@ -0,0 +1,73 @@
+import ROOT
+import sys
+import os
+import math
+import shutil
+from RunKit.sh_tools import sh_call
+if __name__ == "__main__":
+    sys.path.append(os.environ['ANALYSIS_PATH'])
+
+
+import Common.Utilities as Utilities
+from Analysis.HistHelper import *
+
+if __name__ == "__main__":
+    import argparse
+    import yaml
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--histDir', required=True, type=str)
+    parser.add_argument('--test', required=False, type=bool, default=False)
+    parser.add_argument('--remove-files', required=False, type=bool, default=False)
+    parser.add_argument('--dataset', required=True, type=str)
+    parser.add_argument('--sampleConfig', required=True, type=str)
+    args = parser.parse_args()
+
+    sample_cfg_dict = {}
+    with open(args.sampleConfig, 'r') as f:
+        sample_cfg_dict = yaml.safe_load(f)
+
+    if not os.path.isdir(args.histDir):
+        os.makedirs(args.histDir)
+
+    # 1 list files :
+
+    samples_files = {}
+    tmpDir = os.path.join(args.histDir, args.dataset,'tmp')
+
+    sample_type = ""
+    if(args.dataset=='data'): sample_type = 'data'
+    else: sample_type = sample_cfg_dict[args.dataset]['sampleType']
+
+    all_dir_vars = os.listdir(tmpDir)
```

Comment:
- Don't rely on `os.listdir` and `os.walk`
- Don't create txt file
- Don't need to pass dataset and sampleConfig
The signature should be:
```shell
HistProducerSample.py --histDir my/hist/dir --outDir my/out/dir --hists m_tautau,tau1_pt --file-name-pattern 'nano_{id}.root' --file-ids '0-100'
```